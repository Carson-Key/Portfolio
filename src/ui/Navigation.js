import { useState } from 'react'
import { MdMenu } from 'react-icons/md'
import Logo512 from '../assets/Logo-512.png'

const Navigation = () => {
    const [drawerMenuIsOpen, setDrawerMenuIsOpen] = useState(false)

    const openDrawerMenu = () => {
        setDrawerMenuIsOpen(!drawerMenuIsOpen)
    }

	return (
		<nav className="text-white">
            <div className="flex relative w-screen h-14 z-40 bg-secondary">
                <button onClick={openDrawerMenu}><MdMenu className="absolute inset-y-0 left-0 h-14 text-2xl ml-2"/></button>
                <div className="flex justify-center my-auto mx-auto">
                    <h1 className="my-auto text-lg font-helvetica">Carson</h1>
                    <img src={Logo512} className="h-10 ml-1" alt="" />
                    <h1 className="my-auto text-lg font-helvetica ml-1">Key</h1>
                </div>
            </div>
			<ul className={
                "absolute top-0 left-0 z-0" + 
                " transform ease-in-out transition-all duration-500 translate-x-0" +
                " w-screen h-screen divide-y px-2 pt-14" +
                " bg-drawer-menu bg-contain bg-no-repeat bg-right-bottom bg-secondary bg-opacity-90" + 
                (drawerMenuIsOpen ? " translate-x-0" : " -translate-x-full")
            }>
                <li className="px-3">Projects</li>
                <li className="px-3">Work</li>
                <li className="px-3">Eduction</li>
                <li className="px-3">Resume</li>
                <li className="px-3">About</li>
            </ul>
		</nav>
	)
}

export default Navigation
