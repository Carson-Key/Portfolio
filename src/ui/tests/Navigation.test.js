// Packages
import renderer from 'react-test-renderer'
import { BrowserRouter as Router } from 'react-router-dom'
// UI
import Navigation from '../Navigation'

describe('Navigation Component', () => {
    it('Should Render Navigation Component', () => {
        const navigaion = renderer.create(
            <Router>
                <Navigation />
            </Router>
        ).toJSON()
        expect(navigaion).toMatchSnapshot()
    })
})