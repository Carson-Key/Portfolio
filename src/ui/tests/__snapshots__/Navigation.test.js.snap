// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation Component Should Render Navigation Component 1`] = `
<nav
  className="text-white lg:my-auto"
>
  <ul
    className="absolute top-0 left-0 -z-10 transform ease-in-out transition-all duration-500 translate-x-0 w-screen h-screen divide-y px-2 pt-14 text-2xl bg-drawer-menu bg-contain bg-no-repeat bg-right-bottom bg-secondary bg-opacity-90 -translate-x-full lg:static lg:flex lg:transform-none lg:w-auto lg:h-auto lg:divide-y-0 lg:divide-x lg:mr-0 lg:ml-8 lg:p-0 lg:text-3xl lg:bg-none lg:bg-transparent xl:text-4xl 2xl:text-5xl"
  >
    <li
      className="py-2 lg:py-0"
    >
      <a
        href="/Projects"
        onClick={[Function]}
      >
        <p
          className="px-3 lg:px-4"
        >
          Projects
        </p>
      </a>
    </li>
    <li
      className="py-2 lg:py-0"
    >
      <a
        href="/Work"
        onClick={[Function]}
      >
        <p
          className="px-3 lg:px-4"
        >
          Work
        </p>
      </a>
    </li>
    <li
      className="py-2 lg:py-0"
    >
      <a
        href="/Education"
        onClick={[Function]}
      >
        <p
          className="px-3 lg:px-4"
        >
          Education
        </p>
      </a>
    </li>
    <li
      className="py-2 lg:py-0"
    >
      <a
        href="/Resume"
        onClick={[Function]}
      >
        <p
          className="px-3 lg:px-4"
        >
          Resume
        </p>
      </a>
    </li>
    <li
      className="py-2 lg:py-0"
    >
      <a
        href="/About"
        onClick={[Function]}
      >
        <p
          className="px-3 lg:px-4"
        >
          About
        </p>
      </a>
    </li>
  </ul>
</nav>
`;
